### PR TITLE
Add some more aggregate sqllogictests and remove rust tests

### PR DESCRIPTION
**Pull request by mvanschellebeeck**
Ports more aggregate tests from aggregate.rs and removes the existing ones.

Work towards closing https://github.com/apache/arrow-datafusion/issues/4495

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 1]
```diff
@@ -274,187 +271,367 @@ SELECT approx_distinct(c9) AS a, approx_distinct(c9) AS b FROM aggregate_test_10
...
-#  2 8.363636363636363
-#  3 20.789473684210527
-#  4 1.2608695652173914
-#  5 -13.857142857142858
-#    7.81
-
-# TODO: fix decimal places
+query TIR
+SELECT c1, c2, AVG(c3) FROM aggregate_test_100_by_sql GROUP BY CUBE (c1, c2) ORDER BY c1, c2
+----
```
Comment (xudong963) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:461]: Do we still need to print so many lines? How about using `limit` to control the output? cc @mvanschellebeeck @alamb

Comment (xudong963) [reply datafusion/core/tests/sqllogictests/test_files/aggregate.slt:461]: Simplifying during migration or in separate PR both ok to me. -- Seems during migration can reduce our workload

Comment (mvanschellebeeck) [reply datafusion/core/tests/sqllogictests/test_files/aggregate.slt:461]: 👍 - simplified in [41fe83f35300](https://github.com/apache/arrow-datafusion/pull/4505/commits/41fe83f35300d7c628cfce76dc3338c688e510ca)

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 2]
```diff
@@ -274,187 +271,290 @@ SELECT approx_distinct(c9) AS a, approx_distinct(c9) AS b FROM aggregate_test_10
 ----
 100 100
 
-# TODO: csv_query_approx_percentile_cont
+## This test executes the APPROX_PERCENTILE_CONT aggregation against the test
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:274]: 👍  thank you for porting this over

**Review by alamb — APPROVED** [review]
I reviewed the tests that were removed from rs and verified that there were corresponding tests in the .slt file. Thank you @mvanschellebeeck

**Pull request merged**